### PR TITLE
Wait until social networks are ready before populating PYMK.

### PR DESCRIPTION
**Pull request by yipingliao**
@2is10 
Previously, the PYMK button was populated once on module load (before the networks are loaded from the backend), and then once again when the networks are updated. Changing it to just populate once, after the networks have been loaded from the backend.

This bug manifested in the UI as a "Facebook flicker" - it actually flickers every time the first selected button is different from the second selected button.

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/friends/peopleYouMayKnow.js [hunk 1]
```diff
@@ -121,25 +121,20 @@ angular.module('kifi')
 
         function getEligibleNetworksCsv() {
           var onTwitterExperiment = _.indexOf(profileService.me.experiments, 'twitter_beta') > -1;
-          return _.compact([
+          var csv = _.compact([
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/friends/peopleYouMayKnow.js:124]: Hmm... looks like this function always returns `undefined` now.

Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/friends/peopleYouMayKnow.js:124]: Nevermind. I misread the diff. Despite the name, I now see this is no longer a getter, and is not supposed to return anything.

Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/friends/peopleYouMayKnow.js:124]: Yes. Its return value is no longer being used.

Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/friends/peopleYouMayKnow.js:124]: Maybe I should rename it to `initEligibleNetworks`.

File: kifi-backend/modules/shoebox/angular/src/friends/peopleYouMayKnow.js [hunk 2]
```diff
@@ -121,25 +121,20 @@ angular.module('kifi')
...
           ]).join(',');
-        }
 
-        //
-        // Watches and listeners
-        //
-        scope.$watch(getEligibleNetworksCsv, function (csv) {
           scope.network = csv ? _.sample(csv.split(',')) : null;
-        });
+        }
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/friends/peopleYouMayKnow.js:132]: I think the reason for the `$watch` was that if the user connects a network, the page should update. Do we always leave the page when connecting a network now?

An alternative fix would just be to move the `$watch` call inside of a `socialService.refresh().then` callback.

**Comment by 2is10** [comment]
:+1: Thanks for fixing.

File: kifi-backend/modules/shoebox/angular/src/friends/peopleYouMayKnow.js (hunk 2 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/friends/peopleYouMayKnow.js:132]: You're right. Will fix.

**Pull request merged**